src/AWS.Distro.OpenTelemetry.AutoInstrumentation/OtlpUdpExporter.cs

```diff
@@ -0,0 +1,343 @@
+
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Text;
+using Google.Protobuf;
+using Newtonsoft.Json;
+using OpenTelemetry;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+using OpenTelemetry.Proto.Trace.V1;
+using OpenTelemetry.Resources;
+using OtlpResource = OpenTelemetry.Proto.Resource.V1;
+
+/// <summary>
+/// OTLP UDP Exporter class. This class is used to build an OtlpUdpExporter to registered as in exporter
+/// during the instrumentation initialization phase
+/// </summary>
+public class OtlpUdpExporter : BaseExporter<Activity>
+{
+    private UdpExporter udpExporter;
+    private string signalPrefix;
+    private Resource processResource;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OtlpUdpExporter"/> class.
+    /// </summary>
+    /// <param name="endpoint">Endpoint to export requests to</param>
+    /// <param name="signalPrefix">Sampled vs UnSampled signal prefix</param>
+    /// <param name="processResource">Otel Resource object</param>
+    public OtlpUdpExporter(Resource processResource, string? endpoint = null, string? signalPrefix = null)
+    {
+        endpoint = endpoint ?? UdpExporter.DefaultEndpoint;
+        this.udpExporter = new UdpExporter(endpoint);
+        this.signalPrefix = signalPrefix ?? UdpExporter.DefaultFormatOtelTracesBinaryPrefix;
+        this.processResource = processResource;
+    }
+
+    /// <inheritdoc/>
+    public override ExportResult Export(in Batch<Activity> batch)
+    {
+        byte[]? serializedData = this.SerializeSpans(batch);
+        if (serializedData == null)
+        {
+            return ExportResult.Failure;
+        }
+
+        try
+        {
+            this.udpExporter.SendData(serializedData, this.signalPrefix);
+            return ExportResult.Success;
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Error exporting spans: {ex.Message}");
+            return ExportResult.Failure;
+        }
+    }
+
+    /// <inheritdoc/>
+    protected override bool OnShutdown(int timeoutMilliseconds)
+    {
+        try
+        {
+            this.udpExporter.Shutdown();
+            return true;
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Error shutting down exporter: {ex.Message}");
+            return false;
+        }
+    }
+
+    // Function that uses reflection to call ResourceExtensions.ToOtlpResource function.
+    // This functions converts from an OpenTelemetry.Resources.Resource to
+    // OpenTelemetry.Proto.Resource.V1.Resource (protobuf resource to be exported)
+    private OtlpResource.Resource? ToOtlpResource(Resource processResource)
+    {
+        Type? resourceExtensionsType = Type.GetType("OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ResourceExtensions, OpenTelemetry.Exporter.OpenTelemetryProtocol");
+
+        if (resourceExtensionsType == null)
+        {
+            return null;
```

Comment:
Should it log any warn/error for return null or caller will take care of null?

Reply:
In this case, null just represents the fact that it wasn't able to get the class using reflection. It's just there for safety. I can add a log.debug there.

Reply:
Added Logger.LogTrace for those (lowest priority log) in case we want to debug later.